### PR TITLE
Db packages #8

### PR DESCRIPTION
**Pull request by kmiddleton14**

### CODE REVIEW

File: db/models/package.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const db = require('APP/db')
+
+const Package = db.define('package', {
+  description: {
+    type: Sequelize.TEXT,
+    allowNull: false
+  },
+  image: {
+    type: Sequelize.STRING,
+    allowNull: true
```
Comment (csjiang) [inline db/models/package.js]: `allowNull` is true by default.

File: db/models/package.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const Sequelize = require('sequelize')
+const db = require('APP/db')
+
+const Package = db.define('package', {
+  description: {
+    type: Sequelize.TEXT,
+    allowNull: false
+  },
+  image: {
```
Comment (csjiang) [inline db/models/package.js]: As a convention, we're calling the image field `imageURL` and adding a validation to check that it is a URL. 
```javascript 
  imageURL: {
    type: Sequelize.STRING,
    validate: {
      isUrl: true
    }
  }
```

File: db/models/package.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const Sequelize = require('sequelize')
+const db = require('APP/db')
+
+const Package = db.define('package', {
```
Comment (csjiang) [inline db/models/package.js:6]: Package needs a `name` field. We can allow null for custom-created packages, or just (later on) include a setter method that generates a package name from the selected base and add-on(s). By the same logic, we should allow the package `description` to be `null` OR set `defaultValue: ''` for the package `description`.

File: db/models/test/package.test.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+const db = require('APP/db')
+const Package = require('APP/db/models/package')
+const {expect} = require('chai')
+
+describe('package', () => {
+  before('wait for the db', () => db.didSync)
+  const examplePackage = {
+    description: 'Susan McGee',
+    image: 'Sun = fun',
```
Comment (csjiang) [inline db/models/test/package.test.js]: Have this test for the `imageURL` field (adapt for `package`): 
```javascript
  let ourWeatherBase;

  beforeEach(() => {
    ourWeatherBase = WeatherBase.build(exampleWeatherBase)
  })

  it('cannot be created with an invalid image URL', () => {
    ourWeatherBase.imageURL = 'hello world!';
    return ourWeatherBase.validate()
    .then(result => {
      expect(result).to.be.an.instanceOf(Error);
      expect(result.message).to.contain('Validation error: Validation isUrl failed');
    })
  })  
```

File: db/models/test/package.test.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  })
+
+  it('does not allow packateType other than Template and Custom', () => {
+    examplePackage.packageType = 'other'
+    let thePackage = Package.build(examplePackage);
+    thePackage.validate()
+      .then(result => expect(result.errors).to.exist)
+  })
+
+})
```
Comment (csjiang) [inline db/models/test/package.test.js:54]: As convention, we're adding the following hook after each model test suite is completed: 
```javascript
  after(() => 
    Model.truncate({ cascade: true })
    .then(() => console.log('Model table cleared after testing!'))
  )
```

**Review by csjiang — CHANGES_REQUESTED** [review]

**Comment by kmiddleton14** [comment]
@csjiang we've updated our branch with these updates

**Review by csjiang — APPROVED** [review]

**Pull request merged**